### ISSUE TITLE
Update service for WDIO v9

### ISSUE THREAD

**Issue by goosewobbler** [webdriverio-community/wdio-electron-service#533]
WDIO v9 is coming along nicely, there is a bit of work to do in the service around this, since we are dropping Node 16 I think we cut a major version and punt the IPC bridge replacement to v8.

Tasks:
- [x] Publish working pre-release builds 
- [x] Ensure dev docs are up to date
  - Document the fix in https://github.com/webdriverio-community/wdio-electron-service/discussions/667
- [x] Manual test with `wdio-electron-service-example`
- [ ] Manual test with `electron-boilerplate` (can use the [v9 branch](https://github.com/webdriverio/electron-boilerplate/tree/v9))
- [x] Update WDIO peerdep range to ">9.0.0" (after the WDIO v9 release)
- [x] Update WDIO deps to "^9.0.0" (after the WDIO v9 release)
- [ ] Publish full release

https://github.com/webdriverio/webdriverio/issues/10362

**Comment by seanpoulter** [comment]
Is the plan to make a `v8` branch and keep v9 development on `main`?

**Comment by goosewobbler** [comment]
Yes, I'd have the WDIO v9 version on `main` which corresponds to v7 of the service.  I'd probably name the branch `wdio-v8` or `v6` to avoid confusion as the service major versions don't match the WDIO ones.  I'm not planning on backporting anything at the moment, but it could be useful to keep around.  I don't think WDIO v9 will affect the service anywhere near as much as previous major versions did.

**Comment by goosewobbler** [comment]
I created the [`wdio-v8` branch](https://github.com/webdriverio-community/wdio-electron-service/tree/wdio-v8) in case we need to backport anything, `main` will require WDIO v9 when #443 and #666 are merged.

**Comment by goosewobbler** [comment]
Re-opened to keep track of the remaining tasks for the release.

**Comment by goosewobbler** [comment]
@christian-bromann The pre-releases for the service update aren't working due to needing to release the dependent packages, I was considering to release them as `@wdio-electron/types` and `@wdio-electron/utils` for which we need the `@wdio-electron` org, or maybe publishing them under the main org as `@wdio/electron-types` and `@wdio/electron-utils` is better?

**Comment by christian-bromann** [comment]
> Or perhaps publishing them under the main org as `@wdio/electron-types` and `@wdio/electron-utils` is better?

Yes, no need to create another org, let's use `@wdio/` to which you can publish to through the bot.

**Comment by goosewobbler** [comment]
Finally got a working pre-release in `7.0.0-beta.9`.  Tested in `wdio-electron-service-example`.

**Comment by goosewobbler** [comment]
`electron-boilerplate` is still broken on v9 branch, I'll release anyway as the service works in `wdio-electron-service-example`.

**Comment by christian-bromann** [comment]
Let me take a look at the boilerplate, I will ping you if I don't know what is going on.

**Comment by christian-bromann** [comment]
It seems like `7.0.0-beta.11` didn't get pushed to NPM 🤔  However I got the boilerplate working with `7.0.0-beta.10` (there was a bug in Stencil that I fixed but which hasn't been released yet, so reverting that version to `v4.18.3` fixed the build).

**Comment by goosewobbler** [comment]
There's still something up with the release and pre-release workflows, I think the dry run is creating tags for some reason...

https://github.com/webdriverio-community/wdio-electron-service/actions/runs/10463303622/job/28975088177
https://github.com/webdriverio-community/wdio-electron-service/actions/runs/10462686326/job/28973369345

**Comment by goosewobbler** [comment]
Reverted to beta.11 and deleted the unused 7.0.0 tag on github, this time I went straight for the full release instead of a dry run followed by full release, it just attempted to release beta.11 again (instead of 7.0.0) 🤔 

https://github.com/webdriverio-community/wdio-electron-service/actions/runs/10463968883/job/28976808705

**Comment by goosewobbler** [comment]
Reopening until we get the full release done, I think we're missing some steps from the release-it monorepo recipe:

https://github.com/release-it/release-it/blob/main/docs/recipes/monorepo.md

**Comment by goosewobbler** [comment]
Release is done, just need to update `electron-boilerplate`.
